lib/Dialect/FIRRTL/Transforms/Dedup.cpp

```diff
@@ -417,16 +433,10 @@ struct Deduper {
           continue;
         }
         if (auto connect = dyn_cast<ConnectOp>(op)) {
-          // Rewrite the connect to connect all values.
-          auto dst = connect.dest();
-          auto src = connect.src();
-          ImplicitLocOpBuilder builder(connect.getLoc(), connect);
-          // Check which side of the connect was updated.
-          if (newValue == dst)
-            fixupConnect(builder, dst, oldType, src, src.getType());
-          else
-            fixupConnect(builder, dst, dst.getType(), src, oldType);
-          connect->erase();
+          fixupConnect<ConnectOp>(connect, newValue, oldType);
```

Comment:
I think we're missing a `continue`, not a functional problem though.  It might make sense to just switch this to `if-else` chain